core/register_core_types.cpp

```diff
@@ -99,6 +99,9 @@ extern void unregister_variant_methods();
 
 void register_core_types() {
 
+	//consistency check
+	ERR_FAIL_COND(sizeof(Callable) > 16);
+
```

Comment:
An `static_assert` would be better to make it a compiler error instead.

Reply:
Ah nice, I thought static assert was C++17 but there is a C++11 version. Can probably do a separate PR later changing a lot of those we have

Reply:
You can use it #36385 or discard if you already did, I spend 30 minutes searching, and I didn't found more.